src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -50,6 +50,8 @@ def create(self, username, password):
                 url_prefix = 'https' if mgr.get_localized_module_option('ssl') else 'http'
 
                 logger.info('Login successful: %s', username)
+                if not pwd_update_required:
+                    Settings.INITIALIZE_CLUSTER_CREATION = False
```

Comment:
After the cephadm bootstrap, the first time we try to login it leads us to the password change form. And then we will loose our redirect link. if pwd_update_required is true then I don't want to loose our redirect link. That's why I did this.

Reply:
Understood! So in this very case, we need to fix that behaviour, because the initial URL should be preserved across the login stage or stages (no matter it involves changing the default password).

By patching that behaviour in the Auth (back-end) controller, we are mixing front-end and back-end concerns. Keeping the pre-login URL is a front-end concern. It should be the front-end (and the front-end alone) who takes care of keeping the pre-login URL.

Reply:
Created the PR https://github.com/ceph/ceph/pull/41377

Reply:
Thanks!